scripts/suite_objects.py

```diff
@@ -1479,24 +1486,44 @@ def write_var_debug_check(self, var, internal_var, cldicts, outfile, errcode, er
             ubound_string = '(' + ','.join(ubound_strings) + ')'
 
             # Write size check
-            outfile.write(f"if ({conditional}) then", indent)
-            outfile.write(f"! Check size of array {local_name}", indent+1)
-            outfile.write(f"if (size({local_name}{dim_string}) /= {array_size}) then", indent+1)
-            outfile.write(f"write({errmsg}, '(a)') 'In group {self.__group.name} before {self.__subroutine_name}:'", indent+2)
-            outfile.write(f"write({errmsg}, '(2(a,i8))') 'for array {local_name}, expected size ', {array_size}, ' but got ', size({local_name})", indent+2)
-            outfile.write(f"{errcode} = 1", indent+2)
-            outfile.write(f"return", indent+2)
-            outfile.write(f"end if", indent+1)
-            outfile.write(f"end if", indent)
+            if conditional != '.true.':
```

Comment:
You could declare a local `tmp_indent` variable to handle most of these lines in the same code and avoid duplication ... we can discuss in 20 mins.

Reply:
@climbfuji Good idea. See below.